src/OpenTelemetry.Exporter.Geneva/Internal/EventHeader/EventHeaderLogExporter.cs

```diff
@@ -0,0 +1,547 @@
+                }
+            }
+
+            return result;
+        }
+
+        return ExportResult.Failure;
+    }
+
+    public void Dispose()
+    {
+        if (this.isDisposed)
+        {
+            return;
+        }
+
+        try
+        {
+            // DO NOT Dispose eventBuilder, keyValuePairs, and partCFields as they are static
+            this.serializationData?.Dispose();
+        }
+        catch (Exception ex)
+        {
+            ExporterEventSource.Log.ExporterException("EventHeaderLogExporter Dispose failed.", ex);
+        }
+
+        this.isDisposed = true;
+    }
+
+    internal EventHeaderDynamicBuilder SerializeLogRecord(LogRecord logRecord)
+    {
+        IReadOnlyList<KeyValuePair<string, object?>>? listKvp;
+
+        // `LogRecord.State` and `LogRecord.StateValues` were marked Obsolete in https://github.com/open-telemetry/opentelemetry-dotnet/pull/4334
+#pragma warning disable 0618
+        if (logRecord.StateValues != null)
+        {
+            listKvp = logRecord.StateValues;
+        }
+        else
+        {
+            // Attempt to see if State could be ROL_KVP.
+            listKvp = logRecord.State as IReadOnlyList<KeyValuePair<string, object?>>;
+        }
+#pragma warning restore 0618
+
+        // Structured log.
+        // 2 scenarios.
+        // 1. Structured logging with template
+        // eg:
+        // body
+        // "Hello from {food} {price}."
+        // part c
+        // food = onion
+        // price = 100
+        // TODO: 2. Structured with strongly typed logging.
+
+        var categoryName = logRecord.CategoryName ?? this.defaultEventName;
+
+        // If user configured explicit TableName, use it.
+        if (this.tableMappings?.TryGetValue(categoryName, out var eventName) != true)
+        {
+            if (!this.shouldPassThruTableMappings)
+            {
+                eventName = this.defaultEventName;
+            }
+            else
+            {
+                // TODO: Avoid allocation
+                eventName = GetSanitizedCategoryName(categoryName);
+            }
+        }
+
+        var eb = EventBuilder.Value ??= new EventHeaderDynamicBuilder(); // TODO: make sure it can be reused with a reset
+
+        var timestamp = logRecord.Timestamp;
+
+        eb.Reset(eventName!);
+        eb.AddUInt16("__csver__", 1024, Microsoft.LinuxTracepoints.EventHeaderFieldFormat.HexInt);
+
+        eb.AddStructWithMetadataPosition("PartA", out var partAFieldsCountMetadataPosition);
+        Serialize(eb, "time", timestamp); // TODO: this is different from TldLogExporter due to lack of AddFileTime method.
+        byte partAFieldsCount = 1; // TODO: time field is not counted as PartA fields in TldLogExporter. Is it missing? Should time be counted as PartA field?
+        if (this.repeatedPartAFields != null)
+        {
+            eb.AddRawFields(this.repeatedPartAFields.Value);
+        }
+
+        partAFieldsCount += this.partAFieldsCount;
```

Comment:
why it is different from Tld, there it is `var partAFieldsCount = this.partAFieldsCount`

Reply:
Fix it to be the same.